lib/rules/require-array-destructuring.js

```diff
@@ -0,0 +1,29 @@
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+  configure: function(option) {
+    assert(option === true, this.getOptionName() + ' requires a true value');
+  },
+
+  getOptionName: function() {
+    return 'requireArrayDestructuring';
+  },
+
+  check: function(file, errors) {
+    file.iterateNodesByType('VariableDeclaration', function(node) {
+
+      node.declarations.forEach(function(declaration) {
+        if (!declaration.init || declaration.init.type !== 'MemberExpression') {
+          return;
+        }
+
+        var property = declaration.init.property || {};
+        if (property.type === 'Literal' && /^\d+$/.test(property.value)) {
```

Comment:
It would be nice to put a cap on how far into the array we need to do destructuring. I like changing
```js
const foo = foos[0]
```
to
```js
const [ foo ] = foos;
```

I'm even happy with changing
```js
const foo = foos[1]
```
to
```js
const [ , foo ] = foos;
```

But `const [ ,,,,,, foo ] = foos` is pretty hard to read.